### ISSUE TITLE
Search by Owner Original ID

### ISSUE THREAD

**Issue by RickMohr** [OpenTreeMap/otm-core#2734]
A new advanced search option on the "General" tab.

Also, display planting site ID instead of tree ID on plot detail page.

**Comment by RobinIsTheBird** [comment]
Since we had the meeting, we're blocked waiting on Scott's wireframe, and then we need to re-point this issue.

**Comment by RobinIsTheBird** [comment]
Based on the outcome of the trees-team issue 78, the queued issue is now about searching by Owner Original ID.

That issue is spawning a number of new issues, which must be deferred to future sprints.

For this issue, here is what we are going to do, pending approval by @lederer , @RickMohr , and @jwalgran .
1. Make search by Owner Original ID configurable on the management / field configuration / search fields page.
2. Add the option to Advanced Search.
3. Operate as a filter - making non-matching trees transparent.
4. If it resolves to a single tree, select it and zoom, similar to search by location.

If it resolves to multiple trees, it would be nice to zoom to a level that shows just those results, but that is out of scope for this issue.

If a planting site has an Owner Original ID, it presently appears on the detail page. If this feature gets popular, we might consider also showing it in the sidebar detail panel and/or the popup, but that is out of the scope of this issue.

**Comment by RobinIsTheBird** [comment]
The user story, as I understand it from @hadleystein1 , goes something like this:

As a _city manager (or urban forester?)_, I want to _locate a specific tree_, so that I _can tell the crew exactly which tree to service_.

According to Hadley, the manager has a list of trees based on the owner original ID, not from zooming on the open treemap map and selecting.

**Comment by jwalgran** [comment]
This sounds like a workable solution.

I have a question about zooming behavior. If I filter by Owner Original ID and get 1 result, the map zooms and selects. If I filter by a custom field and get 1 result, what happens? Do we also zoom and select?

**Comment by hadleystein1** [comment]
Right now, if you filter by a custom field and get one result that result is highlighted and all other results are grayed out. The single result is not selected and the area is not zoomed in on further. See example below.

![screen shot 2016-08-30 at 9 24 01 pm](https://cloud.githubusercontent.com/assets/15219245/18112767/2668a8ba-6ef8-11e6-9a7d-5f828f6a574e.png)

Is that what you are getting at @jwalgran?

**Comment by lederer** [comment]
I think we should automatically _zoom_ and _select_ any singleton result. This would apply to any search, over any field(s), that results in a single match.

This is particularly important in our current map visual design system – where the circle markers easily blend into the map tiles.

**Comment by dboyer** [comment]
I'm also good with zooming to any single result, no matter the search criteria. That's likely another additional task though.

**Comment by RobinIsTheBird** [comment]
I hate to put provisional items into a sprint goal, but since I'm new to this part of the code, I don't know the relative ease or difficulty of making the select & zoom apply in general vs. for this search alone. I'd rather make headway on #2736  right now than research it.

Unless @RickMohr or @jwalgran can easily tell us the relative ease, I'd say that select/zoom for just this search satisfies the Done criteria for this issue, but we won't object if it works for all search criteria. If it works for all criteria, that will also close #2745 and partially satisfy #2746, in violation of #753.

**Comment by jwalgran** [comment]
I assume that implementing the same zoom behavior regardless of the filter details will be easier than implementing a special case for a single field. Implementing the zooming behavior is correctly separated out as #2745 and is not part of adding searching by Owner Original ID.

**Comment by RobinIsTheBird** [comment]
As Justin points out:

> I assume that implementing the same zoom behavior regardless of the filter details will be easier than implementing a special case for a single field. Implementing the zooming behavior is correctly separated out as #2745 and is not part of adding searching by Owner Original ID

So it makes sense to drop point 5 from the completion criteria.

**Comment by maurizi** [comment]
> A new advanced search option on the "General" tab.

"Owner Original ID" is a field on Planting Sites, not all Map Features, so it'll be on the Planting Site tab.
